### PR TITLE
feat: Add cluster attach or detach key pairs

### PR DESCRIPTION
**Pull request by whDongRui**
* fix: Home detail header show bug

* fix: Purchased page apps query bug

### CODE REVIEW

File: src/components/Header/index.jsx [hunk 1]
```diff
@@ -41,8 +41,12 @@ class Header extends Component {
     return pathname.indexOf(curLink) > -1;
   };
 
-  becomeDeveloper = () => {
-    setCookie('role', '', -1);
+  becomeDeveloper = isNormal => {
+    if (isNormal) {
```
Comment (iwisunny) [inline src/components/Header/index.jsx]: you can use `user provider`'s method

File: src/providers/user.js [hunk 2]
```diff
@@ -26,16 +26,23 @@ export default class UserProvider {
...
+      this.changedRole = getCookie('role');
+      if (this.changedRole === ROLE_NORMAL) {
+        this.canChangeRole();
+      }
+    } catch (err) {}
 
     this.accessToken = getCookie('access_token');
   }
 
   canChangeRole() {
```
Comment (iwisunny) [inline src/providers/user.js]: this method should return boolean

File: src/components/Layout/index.jsx [hunk 3]
```diff
@@ -72,11 +72,11 @@ export default class Layout extends React.Component {
       backBtn,
       hasSearch,
       title,
-      noLogin
```
Comment (iwisunny) [inline src/components/Layout/index.jsx]: this flag will be removed, no need to use

Comment (whDongRui) [reply src/components/Layout/index.jsx]: ok.

File: src/providers/user.js (hunk 2 above)
Comment (whDongRui) [reply src/providers/user.js]: This method for change the judgment of role，rename changeJudgeRole()

File: src/components/Header/index.jsx (hunk 1 above)
Comment (whDongRui) [reply src/components/Header/index.jsx]: The method for add cookie to role when changed normal view, or remove cookie when back developer view.

**Pull request merged**